src/base/static/js/views/map-view.js

```diff
@@ -77,33 +77,57 @@
 
       // Bind visiblity event for custom layers
       $(Shareabouts).on('visibility', function (evt, id, visible, isBasemap) {
-        var layer = self.layers[id],
-        config = _.find(self.options.mapConfig.layers, function(c) {
-          return c.id === id;
-        });
-        if (config && !config.loaded && visible) {
-          self.createLayerFromConfig(config);
-          config.loaded = true;
-          layer = self.layers[id];
-        }
-        if (isBasemap) {
-          _.each(self.options.basemapConfigs, function(basemap) {
-            if (basemap.id === id) {
-              self.map.addLayer(layer);
-              layer.bringToBack();
-            } else if (self.layers[basemap.id]) {
-              self.map.removeLayer(self.layers[basemap.id]);
-            }
-          });
-        } else if (layer) {
-          self.setLayerVisibility(layer, visible);
+        var layerIds;
+
+        // If this layer is a composite of other layers, get the list of
+        // constituent layers to turn off or on. Otherwise, assume the group id
+        // corresponds to the id of the layer to turn off or on.
+        // NOTE: we assume the gis-layers section on the sidebar config is the
+        // first section listed under the panels section.
+        var sidebarLayerItem = _(self.options.sidebarConfig.panels[0].groupings)
+          .chain()
+          .pluck("layers")
+          .flatten()
+          .findWhere({id: id})
+          .value();
+
+        if (sidebarLayerItem && sidebarLayerItem.constituentLayers) {
+          layerIds = sidebarLayerItem.constituentLayers;
         } else {
-          // Handles cases when we fire events for layers that are not yet
-          // loaded (ie cartodb layers, which are loaded asynchronously)
-          // We are setting the asynch layer config's default visibility here to
-          // ensure they are added to the map when they are eventually loaded:
-          config.asyncLayerVisibleDefault = visible;
+          layerIds = [id];
         }
+  
+        _.each(layerIds, function(layerId) {
```

Comment:
Instead of changing this function to always have an array, you might consider making a standard layer the default, and then making a composite layer a special case that fires off a 'visibility' event for every constituent layer. Not sure if you've already thought about that and ruled it out, but I think it would make things a little clearer and remove the need for your big comment above.